### PR TITLE
CI: Move cc_tests workflow to Engflow's remote execution

### PR DESCRIPTION
**Pull request by lfpino**
Signed-off-by: Luis Fernando Pino Duque <luis@engflow.com>

Description:
Moving the cc_tests presubmit workflow to Engflow cuts the build time to almost 10x less

Before (>1hr):
Before_cctests

After (~7min):
After_cctests

Risk Level: Low
Testing: See CI checks (cc_tests)
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -88,20 +88,52 @@ build:coverage --javabase=@bazel_tools//tools/jdk:remote_jdk11
...
+# remote-ci-linux: These options are linux-only
+#############################################################################
+build:remote-ci-linux --config=engflow
+build:remote-ci-linux --config=remote-ci-common
+build:remote-ci-linux --remote_default_exec_properties=Pool=linux
+#############################################################################
+# remote-ci-macos: These options are macOS-only
+#############################################################################
+build:remote-ci-macos --config=remote-ci-common
+build:remote-ci-macos --remote_default_exec_properties=Pool=macos
```
Comment (ulfjack) [inline .bazelrc:136]: This isn't setup yet. That's why you got a timeout.

File: WORKSPACE [hunk 2]
```diff
@@ -46,3 +46,20 @@ declare_python_abi(name = "python_abi", python_version = "3")
...
+)
+
+load("@bazel_toolchains//rules:rbe_repo.bzl", "rbe_autoconfig")
+
+rbe_autoconfig(
+    name = "engflow_remote_config",
+    digest = "sha256:375bf44de0d891f881fd38d7732db411f1f34ec6200eac2f1c9fedf4ad0e474d",
+    registry = "docker.io",
+    repository = "envoyproxy/envoy-build-ubuntu",
+)
```
Comment (ulfjack) [inline WORKSPACE:65]: Add:
```
    exec_properties = {
        "Pool": "linux",
    },
```

**Comment by lfpino** [comment]
This was done in https://github.com/envoyproxy/envoy-mobile/pull/1697

**Pull request closed without merging**